STAR-Request/draft-sheffer-acme-star-delegation.md

```diff
@@ -0,0 +1,302 @@
+- In case the identifier type is "dns", it MAY contain a "cname" field with the alias of the identifier in the NDC domain.  This field is used by the IdO to create the DNS aliasing needed to redirect the resolvers to the delegated entity.
+
+~~~
+   POST /acme/new-order HTTP/1.1
+   Host: acme.dno.example
+   Content-Type: application/jose+json
+
+   {
+     "protected": base64url({
+       "alg": "ES256",
+       "kid": "https://acme.dno.example/acme/acct/evOfKhNU60wg",
+       "nonce": "5XJ1L3lEkMG7tR6pA00clA",
+       "url": "https://acme.dno.example/acme/new-order"
+     }),
+     "payload": base64url({
+       "identifiers": [
+         {
+           "type": "dns",
+           "value": "abc.ndc.dno.example",
+           "delegated": true,
+           "cname": "abc.ndc.example"
+         }
+       ],
+     }),
+     "signature": "H6ZXtGjTZyUnPeKn...wEA4TklBdh3e454g"
+   }
+~~~
+
+The Order object that is created on the IdO:
+
+- MUST start in the "ready" state;
+- MUST contain an empty "authorizations" field;
+- MUST NOT contain the "notBefore" and "notAfter" fields.
+
+~~~
+   {
+     "status": "ready",
+     "expires": "2016-01-01T00:00:00Z",
+
+     "identifiers": [
+      {
+        "type": "dns",
+        "value": "abc.ndc.dno.example",
+        "delegated": true,
+        "cname": "abc.ndc.example"
+      }
+     ],
+
+     "authorizations": [],
+
+     "finalize": "https://acme.dno.example/acme/order/TOlocE8rfgo/finalize"
+   }
+~~~
+
+The IdO SHOULD copy any "recurrent-*" field from the NDC request into the related STAR request to the ACME CA.
+
+When the validation of the identifiers has been successfully completed and the certificate has been issued by the CA, the IdO:
+
+- MUST move its Order resource status to "valid";
+- MUST copy the "certificate" field from the STAR Order;
+
+The latter indirectly includes (via the NotBefore and NotAfter HTTP headers) the renewal timers needed by the NDC to inform its certificate reload logics.
+
+~~~
+   {
+     "status": "valid",
+     "expires": "2016-01-01T00:00:00Z",
+
+     "identifiers": [
+      {
+        "type": "dns",
+        "value": "abc.ndc.dno.example",
+        "delegated": true,
+        "cname": "abc.ndc.example"
+      }
+     ],
+
+     "authorizations": [],
+
+     "finalize": "https://acme.dno.example/acme/order/TOlocE8rfgo/finalize",
+
+     "certificate": "https://acme.ca.example/acme/order/yTyr23sSDg9"
+   }
+~~~
+
+Note that at this point in the flow, the IdO can add the CNAME records to its zone.
+
+### Order Object on the IdO-CA side
+
+The Order from the IdO to the ACME CA SHOULD strip the "cname" attribute from the twin Order from the NDC ({{sec-profile-ndc-to-ido}}), and MUST contain the necessary STAR extensions.  In addition, in order to allow the NDC to download the certificate using the unauthenticated GET, it MUST contain the recurrent-certificate-get attribute and set its value to true.
```

Comment:
With regards the NDC account: even if NDC had an account with the same ACME CA, it couldn't access the IdO's orders without the IdO's private key.  So, there's nothing we can do if the NDC omits the unauth'd GET flag or, equivalently, if the ACME CA only accepts POST-as-GET, because we don't want to share private key material between the delegating and the delegated in the first place.